pkg/schedulercache/resource_info.go

```diff
@@ -16,5 +16,71 @@ limitations under the License.
 
 package schedulercache
 
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/api/core/v1"
+)
+
 type Resource struct {
+	MilliCPU float64
+	Memory   float64
+}
+
+func EmptyResource() *Resource {
+	return &Resource{
+		MilliCPU: 0,
+		Memory:   0,
+	}
+}
+
+func (r *Resource) Clone() *Resource {
+	clone := &Resource{
+		MilliCPU: r.MilliCPU,
+		Memory:   r.Memory,
+	}
+	return clone
+}
+
+var minMilliCPU float64 = 10
+var minMemory float64 = 10 * 1024 * 1024
+
+func NewResource(rl v1.ResourceList) *Resource {
+	cpu := rl[v1.ResourceCPU]
+	mem := rl[v1.ResourceMemory]
+
+	return &Resource{
+		MilliCPU: float64(cpu.MilliValue()),
+		Memory:   float64(mem.Value()),
+	}
+}
+
+func (r *Resource) IsEmpty() bool {
+	return r.MilliCPU < minMilliCPU && r.Memory < minMemory
+}
+
+func (r *Resource) Add(rr *Resource) *Resource {
+	r.MilliCPU += rr.MilliCPU
+	r.Memory += rr.Memory
+	return r
+}
+
+func (r *Resource) Sub(rr *Resource) *Resource {
+	r.MilliCPU -= rr.MilliCPU
+	r.Memory -= rr.Memory
+	return r
+}
+
+func (r *Resource) Less(rr *Resource) bool {
+	return r.MilliCPU < rr.MilliCPU && r.Memory < rr.Memory
+}
+
+func (r *Resource) LessEqual(rr *Resource) bool {
+	return (r.MilliCPU < rr.MilliCPU || math.Abs(rr.MilliCPU-r.MilliCPU) < 0.01) &&
+		(r.Memory < rr.Memory || math.Abs(rr.Memory-r.Memory) < 1)
```

Comment:
suggest remove `Less()` and `LessEqual()` temporarily, or compare `MilliCPU` and `Memory` independently

Reply:
Meaning wherever we do need to compare the two Resources, we do it manually each time?

Reply:
as @k82cn mentioned in above comments, I misunderstand the usage of `Less()` and `LessEqual()`, we can keep the current behavior

Reply:
Ok :) 👍